### ISSUE TITLE
VoltageSwitch does not support negative hysteresis

### ISSUE THREAD

**Issue by Developer-Alexander** [SpiceSharp/SpiceSharp#209]
I was expecting that VoltageSwitch behave like when used with a negative hysteresis:

> If Vh is negative, like in this case, the switch will smoothly transition between the on and off impedances. 

[https://www.analog.com/en/technical-articles/ltspiceiv-voltage-controlled-switches.html](url)

But smooth transition between the on and off impedances when used with negative hysteresis seems to be not implemented.

I want to ask kindly to add this feature.

**Comment by svenboulanger** [comment]
Hi,

I currently don't have the intention of adding LTSpice-only features in the core package (also, LTSpice is closed source so the best I could do is emulate the behavior).

If you would like to achieve the kind of transition that they are talking about, perhaps you might take a look at [SpiceSharp.Behavior](https://github.com/SpiceSharp/SpiceSharpBehavioral)?

Yours sincerely,
Sven

**Comment by Developer-Alexander** [comment]
I don't think that this feature is LTSpice only. ngSpice / Pyspice does support it aswell. And it is a very useful feature in my opinion.

Alternatively: would you mind to make the private members of `SpiceSharp.Components.Switches.Biasing` protected instead? This would allow to overwrite it to implement something like a `SmoothVoltageSwitch`.

**Comment by svenboulanger** [comment]
PySpice is not really a simulator, it does reference ngSpice and Xyce however.

I could not find the implementation in ngSpice ([their loading behavior](https://github.com/ngspice/ngspice/blob/master/src/spicelib/devices/sw/swload.c)). Perhaps they transform a negative hysteresis to a different kind of component? I couldn't find documentation about this in their manual either (v38, maybe too old?). I am not sure about Xyce. I could not find any switches in HSpice and I couldn't find info about negative hysteresis in PSpice, TSpice or SmartSpice.

The alternative part I could look into. I'm generally in favor of exposing as much as possible such that it can be overridden.
